src/plugins/intel_gpu/src/graph/graph_optimizer/handle_reshape.cpp

```diff
@@ -214,34 +214,41 @@ void handle_reshape::run(program& p) {
 
             auto reshape_layout = node->get_output_layout();
             auto target_format = format::get_default_format(reshape_layout.get_rank());
+            auto input_layout = input_node.get_output_layout();
+            auto target_layout = layout({reshape_layout.get_partial_shape(), reshape_layout.data_type, target_format});
 
-            if (!(node->is_output()) && (reshape_layout.format != target_format)) {
-                auto target_layout = layout({reshape_layout.get_partial_shape(), reshape_layout.data_type, target_format});
+            if (!(node->is_output()) && !reshape_layout.compatible(target_layout)) {
                 // when some primitive does an implicit reorder to some other format then we lose the info about pitches
                 // in reshape stage we assume user provides the input vector in bfyx
-                if (!reshape_layout.compatible(target_layout)) {
-                    auto reshape_input = std::make_shared<reorder>("reorder:_reshape_input_" + node->id(),
-                                                                   cldnn::input_info(input_node.id(), input_port),
-                                                                   target_format,
-                                                                   reshape_layout.data_type);
-                    GPU_DEBUG_LOG << "reshape_handler: " << reshape_input->id
-                        << " input_info : " << reshape_input->dependencies().front().to_string() << std::endl;
-                    auto& reshape_input_node = p.get_or_create(reshape_input);
-                    p.add_intermediate(reshape_input_node, *node, 0, reshape_input_node.get_dependencies().empty());
-                    reshape_input_node.recalc_output_layout();
-
-                    auto reshape_users = node->get_users();
-                    for (const auto& user : reshape_users) {
-                        auto reshape_output = std::make_shared<reorder>("reorder:_reshape_output_" + node->id(),
-                                                                        user->id(),
-                                                                        reshape_layout.format,
-                                                                        reshape_layout.data_type);
-                        auto& reshape_output_node = p.get_or_create(reshape_output);
-                        p.add_intermediate(reshape_output_node,
-                                           *user,
-                                           *node,
-                                           reshape_output_node.get_dependencies().empty());
-                        reshape_output_node.recalc_output_layout();
+                if ((reshape_layout.format != target_format) || (reshape_layout.format != input_layout.format)) {
```

Comment:
Did you check benchmark tests?

Reply:
Checked in ADL(http://ikswven004.ikor.intel.com:8080/job/12900k-1-iGPU-FP/80/artifact). No regression.